eng/doc/release-branch-links.json

```diff
@@ -0,0 +1,112 @@
+[
+  {
+    "version": "go1.20",
+    "stable": true,
+    "files": [
+      {
+        "filename": "go1.20.src.tar.gz",
+        "os": "",
+        "arch": "",
+        "version": "go1.20",
+        "kind": "source",
+        "url": "https://aka.ms/golang/release/latest/go1.20.src.tar.gz",
+        "checksumURL": "https://aka.ms/golang/release/latest/go1.20.src.tar.gz.sha256",
+        "signatureURL": "https://aka.ms/golang/release/latest/go1.20.src.tar.gz.sig"
+      },
+      {
+        "filename": "go1.20.linux-amd64.tar.gz",
+        "os": "linux",
+        "arch": "amd64",
+        "version": "go1.20",
+        "kind": "archive",
+        "url": "https://aka.ms/golang/release/latest/go1.20.linux-amd64.tar.gz",
+        "checksumURL": "https://aka.ms/golang/release/latest/go1.20.linux-amd64.tar.gz.sha256",
+        "signatureURL": "https://aka.ms/golang/release/latest/go1.20.linux-amd64.tar.gz.sig"
+      },
+      {
+        "filename": "go1.20.linux-arm64.tar.gz",
+        "os": "linux",
+        "arch": "arm64",
+        "version": "go1.20",
+        "kind": "archive",
+        "url": "https://aka.ms/golang/release/latest/go1.20.linux-arm64.tar.gz",
+        "checksumURL": "https://aka.ms/golang/release/latest/go1.20.linux-arm64.tar.gz.sha256",
+        "signatureURL": "https://aka.ms/golang/release/latest/go1.20.linux-arm64.tar.gz.sig"
+      },
+      {
+        "filename": "go1.20.linux-armv6l.tar.gz",
+        "os": "linux",
+        "arch": "armv6l",
+        "version": "go1.20",
+        "kind": "archive",
+        "url": "https://aka.ms/golang/release/latest/go1.20.linux-armv6l.tar.gz",
+        "checksumURL": "https://aka.ms/golang/release/latest/go1.20.linux-armv6l.tar.gz.sha256",
+        "signatureURL": "https://aka.ms/golang/release/latest/go1.20.linux-armv6l.tar.gz.sig"
+      },
+      {
+        "filename": "go1.20.windows-amd64.zip",
+        "os": "windows",
+        "arch": "amd64",
+        "version": "go1.20",
+        "kind": "archive",
+        "url": "https://aka.ms/golang/release/latest/go1.20.windows-amd64.zip",
+        "checksumURL": "https://aka.ms/golang/release/latest/go1.20.windows-amd64.zip.sha256"
+      }
+    ]
+  },
+  {
+    "version": "go1.19",
+    "oldStable": true,
```

Comment:
Hmm, I added `stable` and `oldStable` with https://github.com/actions/setup-go#using-stableoldstable-aliases in mind, but https://go.dev/dl/?mode=json lists both 1.19.x and 1.20.x as `"stable": true`, so this conflicts a bit.

I think I should just make `stable` always `true`. Then, if we ever put an RC release or something similar in this list, we can make it `false`.

I still think someone might find something like `"latestStable": true` and `"previousStable": true` useful, to avoid having to sort this list, or assume the first in the list is always the latest, or assume that if the latest is 1.31, the previous number is necessarily 1.30. (Maybe in some weird future case we skip 1.30 and still support 1.29+1.31.)